interfaces/builtin/basedeclaration.go

```diff
@@ -178,6 +178,12 @@ slots:
         - core
         - gadget
     deny-auto-connection: true
+  boot-config:
+    allow-installation:
+      slot-snap-type:
+        - gadget
+    deny-connection: true
+    deny-auto-connection: true
```

Comment:
@jdstrand Per comment above, this is not quite the intention. We want to name it boot-control, and hand it to the core snap only. Now that we have core-support, I'm actually wonder if we should even do that or just integrate these settings into core-support itself.

@mvo5 Let's talk today in the standup